modules/stochastic_tools/src/samplers/ActiveLearningMonteCarloSampler.C

```diff
@@ -0,0 +1,89 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ActiveLearningMonteCarloSampler.h"
+#include "Distribution.h"
+
+registerMooseObject("StochasticToolsApp", ActiveLearningMonteCarloSampler);
+
+InputParameters
+ActiveLearningMonteCarloSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Monte Carlo Sampler for active learning with surrogate model.");
+  params.addRequiredParam<dof_id_type>("num_batch",
+                                       "The number of full model evaluations in the batch.");
+  params.addRequiredParam<std::vector<DistributionName>>(
+      "distributions",
+      "The distribution names to be sampled, the number of distributions provided defines the "
+      "number of columns per matrix.");
+  params.addRequiredParam<ReporterName>("flag_sample",
+                                        "Flag samples if the surrogate prediction was inadequate.");
+  params.addParam<unsigned int>(
+      "num_random_seeds",
+      100000,
+      "Initialize a certain number of random seeds. Change from the default only if you have to.");
+  return params;
+}
+
+ActiveLearningMonteCarloSampler::ActiveLearningMonteCarloSampler(const InputParameters & parameters)
+  : Sampler(parameters),
+    ReporterInterface(this),
+    _distribution_names(getParam<std::vector<DistributionName>>("distributions")),
+    _flag_sample(getReporterValue<std::vector<bool>>("flag_sample")),
+    _num_random_seeds(getParam<unsigned int>("num_random_seeds")),
+    _step(getCheckedPointerParam<FEProblemBase *>("_fe_problem_base")->timeStep())
+{
+  for (const DistributionName & name : _distribution_names)
+    _distributions.push_back(&getDistributionByName(name));
+  setNumberOfRows(getParam<dof_id_type>("num_batch"));
+  setNumberOfCols(_distributions.size());
+  _inputs_sto.resize(getParam<dof_id_type>("num_batch"));
+  _inputs_gp_fails.resize(getParam<dof_id_type>("num_batch"));
+  for (unsigned int i = 0; i < _inputs_sto.size(); ++i)
+  {
+    _inputs_sto[i].resize(_distributions.size());
+    _inputs_gp_fails[i].resize(_distributions.size());
+  }
+  _check_step = 0;
+  setNumberOfRandomSeeds(_num_random_seeds);
```

Comment:
Remove the `_num_random_seeds` member variable and do this:
```suggestion
  setNumberOfRandomSeeds(getParam<unsigned int>("num_random_seeds"));
```